pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -1438,7 +1438,7 @@ func (s *Service) getHealthCheckTarget() string {
 	protocol := &infrav1.ELBProtocolSSL
 	if controlPlaneELB != nil && controlPlaneELB.HealthCheckProtocol != nil {
 		protocol = controlPlaneELB.HealthCheckProtocol
-		if protocol == &infrav1.ELBProtocolHTTP || protocol == &infrav1.ELBProtocolHTTPS {
+		if *(protocol) == infrav1.ELBProtocolHTTP || *(protocol) == infrav1.ELBProtocolHTTPS {
```

Comment:
But the tests are passing that you added. :D So what is this solving? The protocol does have a `String` method.

Also the `*()` could just be `*` but honestly, I would rather not have this blown up:

```go
if protocol.String() == infrav1.ELBProtocolHTTP.String() || protocol.String() == infrav1.ELBProtocolHTTPS.String() {
```

Instead then? But again, why are the tests working then? :D

Reply:
Test is passing because the is forcing the same variable address. it is being set in 
```
			&infrav1.AWSLoadBalancerSpec{
				HealthCheckProtocol: infrav1.ELBProtocol("HTTP"),
			},
```
So then the addresses match. But is you set something like 

```
	http := infrav1.ELBProtocol("HTTP")
	&infrav1.AWSLoadBalancerSpec{
		HealthCheckProtocol: &http,
	},
```

then test would fail. I'd expect this to be comparing strings not variables addressees so either `*protocol == infrav1.ELBProtocolHTTP` or `protocol.String() == infrav1.ELBProtocolHTTP.String()` makes sense to me.

Reply:
Let's use the `.String()` method instead, it looks a lot cleaner to me. Thank you both for the comments :)

Reply:
Done! Please take a look at your convenience @Skarlso

Reply:
To prove the value, can we please add a testCase within `TestGetHealthCheckProtocol` where we don't set the API input as the built-in variable as in my example above?

Reply:
This test should have captured the error https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0ca6090bfb6f5a61b4a52b2c9cb9eb7c23d86a32/pkg/cloud/services/elb/loadbalancer_test.go#L2332

so i'm very confused.

Reply:
I've updated the tests